client/src/pages/TicketCreation.js

```diff
@@ -1,11 +1,202 @@
-import React from 'react';
+import React, { useState } from "react";
+
+const WorkOrderForm = () => {
+  const [id, setId] = useState("");
+  const [description, setDescription] = useState("");
+  const [status, setStatus] = useState("");
+  const [location, setLocation] = useState("");
+  const [request_date, setRequestDate] = useState("");
+  const [shop, setShop] = useState("");
+  const [priority, setPriority] = useState("");
+  const [error, setError] = useState(null);
+  const [success, setSuccess] = useState(null);
+
+  // Handle form submission
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    setError(null);
+    setSuccess(null);
+
+    const formData = {
+      id,
+      description,
+      status,
+      location,
+      request_date,
+      shop,
+      priority,
+    };
+
+    try {
+      const response = await fetch("api/searchkey/submit-form", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      });
+
+      const data = await response.json();
+
+      if (!response.ok) {
+        throw new Error(data.error || "An error occurred");
+      }
+
+      setSuccess(data.message);
+
+      resetForm();
+    } catch (err) {
+      setError(err.message || "An error occurred");
+    }
+  };
+
+  // Handle form reset
+  const resetForm = () => {
+    setId("");
+    setDescription("");
+    setStatus("");
+    setLocation("");
+    setRequestDate("");
+    setShop("");
+    setPriority("");
+  };
 
-const TicketCreation = () => {
   return (
-    <div>
-      <h1>This is the TicketCreation page</h1>
+    <div className="container w-50">
+      <h2 className="my-4 text-center">Create Work Order</h2>
+
+      {error && <div className="alert alert-danger">{error}</div>}
+      {success && <div className="alert alert-success">{success}</div>}
+
+      <form onSubmit={handleSubmit}>
+        <div className="mb-3">
+          <label htmlFor="id" className="form-label fw-bold">
+            ID <span className="text-danger">*</span>
+          </label>
+          <input
+            type="text"
+            className="form-control "
+            id="id"
+            value={id}
+            onChange={(e) => setId(e.target.value)}
```

Comment:
I'm not sure if we should allow user to enter in the 'id'. The 'id' attribute is unique key in database and user could enter id that has already stored in the database.